### PR TITLE
(2267) Add overseas routes to recognition fields

### PR DESCRIPTION
**Pull request by Gweaton**
# Changes in this PR

Re-adds summary of recognition routes and link to more information for professionals from other countries, which were removed a couple of weeks ago while we figured out the best way to collect and present this information to users. Also adds mandatory radio buttons for selecting the route itself.

#### Qualification page

image

#### Check your answers

image

#### Viewing a profession

image

#### Qualification page

image

#### Check your answers page

image

#### Viewing a profession

image

### CODE REVIEW

File: src/i18n/en/professions.json [hunk 1]
```diff
@@ -202,6 +205,12 @@
       "ukRecognition": "Routes to recognition within the UK",
       "ukRecognitionUrl": "More about recognition within the UK",
       "otherCountriesRecognition": {
+        "routes": {
+          "label": "Recognition for overseas professionals",
+          "all": "All - there are routes to recognition for professionals from all countries",
+          "some": "Some - qualifications from some countries may be recognised, but professionals from other countries will have to re-qualify",
+          "none": "None - all professionals must re-qualify"
```
Comment (bryangaledxw) [inline src/i18n/en/professions.json:212]: I wonder if this would be nicer formatted like:
image
Though understand if this is unwanted scope creep!

**Review by bryangaledxw — APPROVED** [review]

File: src/i18n/en/professions.json (hunk 1 above)
Comment (Gweaton) [reply src/i18n/en/professions.json:212]: This text is actually for displaying on "show" pages, rather than in the radio buttons, as Al felt we could be more succinct with admin questions but should be more verbose for the public pages. I agree with the sentiment though!

File: src/professions/admin/dto/qualifications.dto.ts [hunk 2]
```diff
@@ -26,4 +26,14 @@ export class QualificationsDto {
   @Transform(({ value }) => preprocessUrl(value))
   @ValidateIf((e) => e.ukRecognitionUrl)
   ukRecognitionUrl: string;
+
+  otherCountriesRecognitionSummary: string;
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply src/professions/admin/dto/qualifications.dto.ts:37]: I think this is independent from the URL, or optional enough that it probably doesn't need enforcing, as it could be an alternative to providing a summary.

File: src/qualifications/presenters/qualification.presenter.ts [hunk 3]
```diff
@@ -24,6 +24,13 @@ export default class QualificationPresenter {
     this.qualification && this.qualification.ukRecognitionUrl,
   );
 
+  readonly otherCountriesRecognitionSummary =
+    this.qualification && this.qualification.otherCountriesRecognitionSummary;
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply src/qualifications/presenters/qualification.presenter.ts]: Good point. Have added this in, and made the UK recognition text input the same (a multiline textarea).

File: views/admin/professions/check-your-answers.njk [hunk 4]
```diff
@@ -307,6 +307,23 @@
                   ]
                 }
               } if not isUK else undefined,
+              {
+                key: {
+                  text: ("professions.form.label.qualifications.otherCountriesRecognition.routes.label" | t)
+                },
+                value: {
+                  text: (qualification.adminSelectedOtherCountriesRecognitionRoutes or "") | t
```
Comment (bryangaledxw) [inline views/admin/professions/check-your-answers.njk:315]: Can this be `(qualification.adminSelectedOtherCountriesRecognitionRoutes | t) if qualification.adminSelectedOtherCountriesRecognitionRoutes else ""`, so we're not knowingly passing an invalid ID into the translation filter?

**Review by bryangaledxw — APPROVED** [review]
Looks good, just one more thing I spotted

File: views/admin/professions/check-your-answers.njk (hunk 4 above)
Comment (Gweaton) [reply views/admin/professions/check-your-answers.njk]: As discussed, have made the presenter method return the raw value, which we append to the localisation id here in the template, as nunjucks seems to evaluate everything in an `if` oneliner for some reason...
